src/elastio-snap.c

```diff
@@ -3455,10 +3435,11 @@ static int memory_is_too_low(struct snap_device *dev) {
 		totalram = si.totalram;
 	}
 
-	ret = ((si_mem_available() * 100) / totalram) < LOW_MEMORY_FAIL_PERCENT ? -ENOMEM : tracer_read_memory_fail_state(dev);
-	if (ret && tracer_read_memory_fail_state(dev) == 0) {
+	ret = tracer_read_fail_state(dev);
+	if (ret != -ENOMEM && ((si_mem_available() * 100) / totalram) < LOW_MEMORY_FAIL_PERCENT) {
```

Comment:
This change is intentional. In general driver's code doesn't preserve an error value. I mean, there are no checks like `if (err == 0) {err = SOME_ERR_CODE;}`.
And in case on an error, we'll see all the errors in the `dmesg`. And if driver starts to leak by memory, I'd like to see this second error in `dmesg` too.
But there is no need to call `si_mem_available()` in case if an error is already `ENOMEM`.

Reply:
May be in such case do
```
if (ret != -ENOMEM && ...) {
    LOG_WARN(...)
    if (ret == 0) {
        ret = -ENOMEM;
        tracer_set_fail_state(dev, ret);
    }
}
```
For print message and avoid change cause error state?